### ISSUE TITLE
Different results for "names" after sc.get.rank.genes.df in Jupyter

### ISSUE THREAD

**Issue by malumbres** [scverse/scanpy#1426]
- [x] I have checked that this issue has not already been reported.
- [x] I have confirmed this bug exists on the latest version of scanpy.
- [ ] (optional) I have confirmed this bug exists on the master branch of scanpy.

---

After running sc.tl.rank_genes_groups and sc.get.rank_genes_groups_df using the same data and python/scanpy code, Spider and Jupyter (both installed in the same environment of Anaconda) generate two different values for the "names" column. Whereas Spider gives the expected adata.var_names (e.g. Rpl5; see below), Jupyter gives a numerical code (15721, which is not included in adata.var)

### Minimal code sample (that we can copy&paste without having any data)

In Spider:
```python
de_df.head(5)
```
```pytb
Out[34]: 
     scores  names  logfoldchanges         pvals     pvals_adj
0  9.194006  Rpl15        0.815534  3.784770e-20  1.006711e-15
1  8.427418  Rps28        0.653911  3.533771e-17  4.699739e-13
2  7.989542  Rps21        0.676462  1.354418e-15  1.200872e-11
3  7.871397  Rps27        0.483027  3.507037e-15  2.055341e-11
4  7.859277  Rps24        0.507071  3.863569e-15  2.055341e-11
```

In Jupyter:
```python
de_df.head(5)
```
```pytb
Out[34]: 
  | scores | names | logfoldchanges | pvals | pvals_adj
9.194006 | 15721 | 0.815534 | 3.784770e-20 | 1.006711e-15
8.427418 | 23746 | 0.653911 | 3.533771e-17 | 4.699739e-13
7.989542 | 3910 | 0.676462 | 1.354418e-15 | 1.200872e-11
7.871397 | 5571 | 0.483027 | 3.507037e-15 | 2.055341e-11
7.859277 | 15774 | 0.507071 | 3.863569e-15 | 2.055341e-11

In both cases, Spider and Jupyter
```python
adata.var_names
```
```pytb
Index(['Xkr4', 'Gm1992', 'Gm37381', 'Rp1', 'Sox17', 'Gm37323', 'Mrpl15',
       'Lypla1', 'Gm37988', 'Tcea1',
```

If I try to specify a different column in Jupyter I get this

```python
de_df = sc.get.rank_genes_groups_df(database, group=groupA, gene_symbols="symbol")
```
```pytb
You are trying to merge on object and int64 columns. If you wish to proceed you should use pd.concat
```

#### Versions
[Paste the output of scanpy.logging.print_versions() leaving a blank line after the details tag]
In Spider:
scanpy==1.5.1 anndata==0.7.4 umap==0.4.6 numpy==1.18.5 scipy==1.5.0 pandas==1.0.5 scikit-learn==0.23.1 statsmodels==0.11.1 python-igraph==0.7.1 leidenalg==0.7.0

In Jupyter:
scanpy==1.5.1 anndata==0.7.4 umap==0.4.6 numpy==1.18.5 scipy==1.5.0 pandas==1.0.5 scikit-learn==0.23.1 statsmodels==0.11.1 python-igraph==0.7.1 leidenalg==0.7.0
</details>

**Comment by giovp** [comment]
Hi,

First, the error that you are reporting has to do with series types of the dataframes. Howerver, it's very difficult to provide inputs, because it's unclear what `database` and `groupA` are. Can you report a reproducible example? Also, can you update to scanpy 1.6?

Thank you

**Comment by malumbres** [comment]
Fixed after updating to 1.6! Many thanks!
marcos
